src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -1054,6 +1120,258 @@ get_rebalance_progress(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * WorkerShardSize returns the size of a shard in bytes on a worker, based on
+ * the workerShardStatisticsHash.
```

Comment:
nit; returns `0` if the shard is not on the worker

At least that is what I read from the code. Is it expected shards could be missing? if they are not expected to be missing logging a warning/error might make more sense compared to returning `0`.

Reply:
It is expected, this happens for target shards of moves that have not started yet. Because we don't add those shards to the query since we know they are not there, but we do fetch them from the hash when we display the progress. 

I'll amend the comment to reflect this.